src/utils.ts

```diff
@@ -1198,4 +1352,786 @@ export class RubberBand {
+    path.moveTo(this.origin.x - this.DRAW_START_OFFSET * notVert, this.origin.y - this.DRAW_START_OFFSET * isVert);
+    path.lineTo(this.end.x, this.end.y);
+    path.addPath(endArrow.path);
+    return path
+  }
+
+  private buildPath(): Path2D {
+    const path = new Path2D();
+    const offset = new Vertex(this.DRAW_START_OFFSET * Number(this.isVertical), this.DRAW_START_OFFSET * Number(!this.isVertical));
+    const origin = new Vertex(this.origin.x, this.origin.y).subtract(offset.multiply(2));
+    const size = this.end.subtract(origin).add(offset);
+    path.rect(origin.x, origin.y, size.x, size.y);
+    return path
+  }
+
+  public absoluteToRelative(value: number): number {
+    return this.isVertical ? (value - this.transformMatrix.f) / this.transformMatrix.d : (value - this.transformMatrix.e) / this.transformMatrix.a
+  }
+
+  public relativeToAbsolute(value: number): number {
+    return this.isVertical ? value * this.transformMatrix.d + this.transformMatrix.f : value * this.transformMatrix.a + this.transformMatrix.e
+  }
+
+  private computeMinMax(vector: any[]): number[] {
+    let newVector = vector;
+    if (this.isDiscrete) {return [0, this.labels.length]};
+    return [Math.min(...newVector), Math.max(...newVector)]
+  }
+
+  private computeTicks(): number[] {
+    const increment = newAxis.nearestFive((this.maxValue - this.minValue) / this.nTicks);
+    const remainder = this.minValue % increment;
+    let ticks = [this.minValue - remainder];
+    while (ticks.slice(-1)[0] <= this.maxValue) {ticks.push(ticks.slice(-1)[0] + increment)};
+    if (ticks.slice(0)[0] < this.minValue) {ticks.splice(0, 1)};
+    if (ticks.slice(-1)[0] > this.maxValue) {ticks.splice(-1, 1)};
+    return ticks
+  }
+
+  public draw(context: CanvasRenderingContext2D) {
+    context.lineWidth = this.lineWidth;
+    let color = this.strokeStyle;
+    if (this.mouseStyleON) {color = this.isHover ? this.hoverStyle : this.isClicked ? this.clickedStyle : this.strokeStyle};
+    context.strokeStyle = color;
+    context.fillStyle = color;
+    context.stroke(this.drawPath);
+    context.fill(this.drawPath);
+
+    const canvasHTMatrix = context.getTransform();
+    const pointHTMatrix = canvasHTMatrix.multiply(this.transformMatrix);
+    context.setTransform(pointHTMatrix);
+    this.ticksCoords = this.drawTicks(context, pointHTMatrix, color);
+    context.resetTransform();
+    this.drawName(context, canvasHTMatrix);
+    context.setTransform(canvasHTMatrix);
+    this.drawRubberBand(context);
+  }
+
+  private drawName(context: CanvasRenderingContext2D, canvasHTMatrix: DOMMatrix) {
+    let baseline = ['hanging', 'alphabetic'][Math.sign(1 - Math.sign(this.transformMatrix.f))]
+    let nameCoords = this.end.add(this.origin).divide(2);
+    if (this.isVertical) {nameCoords.x += this.OFFSET_NAME.x ; baseline = ['alphabetic', 'hanging'][Math.sign(1 - Math.sign(this.transformMatrix.e))]}
```

Comment:
On github tool, it is not clear if this line is written on one or several. If it is a one-liner, it might be a better code layout if it was splitted on several lines

Reply:
Splitted